### PR TITLE
Fixes #24651 - Cannot remove packages from content host UI

### PR DESCRIPTION
**Pull request by jturel**
To reproduce:

- Set up a content host with some installed packages
- From the installed package list on the host, select one or more & submit
- Observe error about invalid package name

Leaving some inline comments because I'm making some assumptions here...

### CODE REVIEW

File: app/controllers/katello/api/v2/host_packages_controller.rb [hunk 1]
```diff
@@ -102,13 +102,9 @@ def find_host
...
     end
 
-    def valid_package_name?(package_name)
-      package_name =~ /^[a-zA-Z0-9\-\.\_\+\,]+$/
-    end
-
     def validate_package_list_format(packages)
       packages.each do |package|
-        if !valid_package_name?(package) && !package.is_a?(Hash)
+        unless package.kind_of?(String) && ::Katello::Util::Package.valid_package_name_format(package).nil?
```
Comment (jturel) [inline app/controllers/katello/api/v2/host_packages_controller.rb]: This controller change wasn't necessary but I think it's a little more readable - and got to reuse the Util method

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/content/content-host-packages-installed.controller.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ angular.module('Bastion.content-hosts').controller('ContentHostPackagesInstalled
         var packagesNutupane;
 
         $scope.removeSelectedPackages = function () {
-            var selected = $scope.table.getSelected();
+            var selected = _.map($scope.table.getSelected(), 'name');
```
Comment (jturel) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/content/content-host-packages-installed.controller.js]: The old implementation would take anything and everything from the table and pass it along to the API. The API docs for the package param indicate an array of strings which are package names - so making that consistent.

File: engines/bastion_katello/test/content-hosts/content/content-host-packages-installed.controller.test.js [hunk 3]
```diff
@@ -66,7 +66,7 @@ describe('Controller: ContentHostPackagesInstalledController', function() {
 
         spyOn(HostPackage, 'remove');
         $scope.removeSelectedPackages();
-        expect(HostPackage.remove).toHaveBeenCalledWith({id: mockHost.id, packages: [mockPackageClone]},
+        expect(HostPackage.remove).toHaveBeenCalledWith({id: mockHost.id, packages: ['foo']},
```
Comment (jturel) [inline engines/bastion_katello/test/content-hosts/content/content-host-packages-installed.controller.test.js]: the `mockPackageClone` here contains various NVRA information in different fields which are not in the table today - which has whole `nvra`, `nevra` attributes - I'm not sure if the package table held data in that format before, but now the test matches the behavior I think we really intend.

File: test/controllers/api/v2/host_packages_controller_test.rb [hunk 4]
```diff
@@ -95,6 +95,18 @@ def test_remove
       assert_response :success
     end
 
+    def test_invalid_package_input
```
Comment (jturel) [inline test/controllers/api/v2/host_packages_controller_test.rb]: A test to document what the behavior should be.

**Comment by jturel** [comment]
[test katello]

**Comment by stbenjam** [comment]
[test katello/webpack]

File: app/controllers/katello/api/v2/host_packages_controller.rb [hunk 5]
```diff
@@ -102,13 +102,9 @@ def find_host
...
     end
 
-    def valid_package_name?(package_name)
-      package_name =~ /^[a-zA-Z0-9\-\.\_\+\,]+$/
-    end
-
     def validate_package_list_format(packages)
       packages.each do |package|
-        if !valid_package_name?(package) && !package.is_a?(Hash)
+        unless package.is_a?(String) && ::Katello::Util::Package.valid_package_name_format(package).nil?
```
[the remark this answers is not in the document]
Comment (jturel) [reply app/controllers/katello/api/v2/host_packages_controller.rb]: Agree x 1000! I *only* like to use it when the expression is framed positively as in this case. I think I'll leave this one, but be assured I'd never do something like `unless !foo && !bar` :)

**Pull request merged**